tests/counter/counter_test.go

```diff
@@ -0,0 +1,91 @@
+package counter_test
+
+import (
+	"math"
+	"math/big"
+	"testing"
+
+	"github.com/Fantom-foundation/go-opera/tests"
+	"github.com/Fantom-foundation/go-opera/tests/contracts/counter"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+)
+
+func TestCounter(t *testing.T) {
+	net, err := tests.StartIntegrationTestNet(t.TempDir())
+	if err != nil {
+		t.Fatalf("Failed to start the fake network: %v", err)
+	}
+	defer net.Stop()
+
+	t.Run("can increment and read counter from head", func(t *testing.T) {
+
+		// Deploy the counter contract.
+		contract, _, err := tests.DeployContract(net, counter.DeployCounter)
+		if err != nil {
+			t.Fatalf("failed to deploy contract; %v", err)
+		}
```

Comment:
why not do it before the `t.Run` and use `contract` and `receipt` for both test cases ?

Reply:
Because then, changes done in the account would leak to the other test. This way the test is stateless.